### ISSUE TITLE
Dependency tree error with Payload 1.7.5

### ISSUE THREAD

**Issue by brachypelma** [payloadcms/plugin-search#6]
When trying to install this plugin with the latest version of Payload, I received a dependency tree error.

Steps to reproduce:

1. Run `npx create-payload-app` (using blank template)
2. `cd` into payload app
3. Run `npm i @payloadcms/plugin-search`

Terminal output:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: test@1.0.0
npm ERR! Found: payload@1.7.5
npm ERR! node_modules/payload
npm ERR!   payload@"^1.7.5" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer payload@"^0.18.5" from @payloadcms/plugin-search@1.0.0
npm ERR! node_modules/@payloadcms/plugin-search
npm ERR!   @payloadcms/plugin-search@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /home/eric/.npm/_logs/2023-05-07T14_06_01_015Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/eric/.npm/_logs/2023-05-07T14_06_01_015Z-debug-0.log
```

Here is the `package.json` for my app:

```
{
  "name": "test",
  "description": "Payload project created from blank template",
  "version": "1.0.0",
  "main": "dist/server.js",
  "license": "MIT",
  "scripts": {
    "dev": "cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts nodemon",
    "build:payload": "cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload build",
    "build:server": "tsc",
    "build": "npm run copyfiles && npm run build:payload && npm run build:server",
    "serve": "cross-env PAYLOAD_CONFIG_PATH=dist/payload.config.js NODE_ENV=production node dist/server.js",
    "copyfiles": "copyfiles -u 1 \"src/**/*.{html,css,scss,ttf,woff,woff2,eot,svg,jpg,png}\" dist/",
    "generate:types": "cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload generate:types",
    "generate:graphQLSchema": "cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload generate:graphQLSchema"
  },
  "dependencies": {
    "payload": "^1.7.5",
    "dotenv": "^8.2.0",
    "express": "^4.17.1"
  },
  "devDependencies": {
    "@types/express": "^4.17.9",
    "cross-env": "^7.0.3",
    "nodemon": "^2.0.6",
    "ts-node": "^9.1.1",
    "copyfiles": "^2.4.1",
    "typescript": "^4.8.4"
  }
}
```

I was able to install with the `--legacy-peer-deps` flag, but this is less than ideal.

**Comment by jacobsfletch** [comment]
Hey @brachypelma we are actively working through the react dependencies conflicts you're describing in Payload core. You can track that progress here: https://github.com/payloadcms/payload/pull/2656. Glad you found the  `--legacy-peer-deps` flag, though, that is the temporary workaround. You can also install with `yarn` to have them automatically resolved.

**Comment by richadr** [comment]
Hi @jacobsfletch, this issue is due to peerDependencies of *this* package requiring `payload@^0.18.5`.  This package works fine with payload 1.x, so you could update the range to `^0.18.5 || ^1.0.0`.

**Comment by jacobsfletch** [comment]
@richadr ahh yea I see that now, good catch 👍

**Comment by jacobsfletch** [comment]
@brachypelma @richadr I just released a fix for this in `v1.0.1` 🎉 Let me know how it goes!

**Comment by richadr** [comment]
Tried it immediately, it works, thanks for the swift PR
